src/core/level_selection_system.h

```diff
@@ -14,19 +16,32 @@ class LevelSelectionSystem : public engine::System
     ModelValue mLevelDisplayNamesModel;
     ModelValue mLevelThumbnailsModel;
     std::string mSelectedLevel;
-    std::vector<std::string> mLevelDisplayNames;
-    std::vector<std::string> mLevelRealNames;
-    std::vector<int32_t> mLevelThumbnails;
+    // caches selected game mode from ProgramState
+    std::string mGameMode;
```

Comment:
probs a gamemode enum would be preferable, with a class for enum-string conversions that only happen between json <-> code, so we get an assert / log / whatever, if an invalid value is entered